### PR TITLE
 	Bug 1193923 - Initial ability to import reference data from another server

### PR DESCRIPTION
**Pull request by wlach**
Ignore the second commit, it will disappear when bug 1193925 is merged.

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/869)

### CODE REVIEW

**Comment by maurodoglio** [comment]
You can call get_or_create like this:
```python
instance, created = Model.objects.get_or_create(name=params.pop('name'), defaults=params)
```
where `name` is a field that you can use to retrieve your object and `defaults` is a dict used to create your object if it doesn't exist.

**Comment by wlach** [comment]
Updated per review suggestions. Much more concise now!

File: treeherder/model/management/commands/import_reference_data.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    def handle(self, *args, **options):
+        server_params = urlparse(options['server'])
+        c = TreeherderClient(protocol=server_params.scheme,
+                             host=server_params.netloc)
+
+        # options / option collection hashes
+        for (uuid, props) in c.get_option_collection_hash().iteritems():
+            for prop in props:
+                (option, _) = Option.objects.get_or_create(name=prop['name'])
```
Comment (maurodoglio) [inline treeherder/model/management/commands/import_reference_data.py]: You don't need parenthesis here

File: treeherder/model/management/commands/import_reference_data.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                    option_collection_hash=uuid,
+                    option=option)
+
+        # machine platforms
+        for machine_platform in c.get_machine_platforms():
+            MachinePlatform.objects.get_or_create(
+                    os_name=machine_platform['os_name'],
+                    platform=machine_platform['platform'],
+                    architecture=machine_platform['architecture'],
+                    active_status=machine_platform['active_status']
```
Comment (maurodoglio) [inline treeherder/model/management/commands/import_reference_data.py]: If the key here is the tuple (os_name, platform, architecture) and you leave the active_status field here you can potentially have 2 identical platforms, one active and the other not. If you don't want this behaviour, you can pass `{'active_status': machine_platform['active_status']}` as `defaults`

**Comment by maurodoglio** [comment]
Last 2 comments :stuck_out_tongue:

**Pull request merged**